lib/moment/getMomentAdvancedInfo.ts

```diff
@@ -0,0 +1,43 @@
+import { Moment, MomentAdvancedInfo } from "@/types/moment";
+import selectSale from "@/lib/supabase/in_process_sales/selectSale";
+import getMomentOnChainInfo from "@/lib/viem/getMomentOnChainInfo";
+import { convertDatabaseSaleToApi } from "@/lib/sales/convertDatabaseSaleToApi";
+import { convertOnChainSaleToApi } from "@/lib/sales/convertOnChainSaleToApi";
+import selectMoments from "@/lib/supabase/in_process_moments/selectMoments";
+
+export const getMomentAdvancedInfo = async (moment: Moment): Promise<MomentAdvancedInfo> => {
+  const moments = await selectMoments(moment);
+
+  const momentdata = moments[0];
+  if (momentdata) {
+    const uri = momentdata.uri;
+    const owner = momentdata.collection.default_admin;
+    const sale = await selectSale(momentdata.id);
+
+    if (sale) {
+      return {
+        uri,
+        owner,
+        saleConfig: convertDatabaseSaleToApi(sale),
+      };
+    }
+
+    const { saleConfig: onChainSale } = await getMomentOnChainInfo(moment);
+    return {
+      uri,
+      owner,
+      saleConfig: convertOnChainSaleToApi(onChainSale),
+    };
+  }
+
+  const {
+    saleConfig: onChainSale,
+    owner: tokenOwner,
+    tokenUri,
+  } = await getMomentOnChainInfo(moment);
```

Comment:
Same feedback here.
- Why is the front-end repo reading from onchain data directly?
- shouldn't our indexer be storing the latest info directly in supabase?